### PR TITLE
Update ESLint shared config to v3

### LINKED ISSUE #8982: Update MetaMask ESLint config to v2
This issues serves as a tracking issue for various ESLint config update work.

### PR DESCRIPTION
**Pull request by whymarrh**
Closes #8982

This PR updates the ESLint config to the latest version, v3.

There are two commits with fixes, one that's automated and one that was manual. Only the first and last commit require review.

I disabled the following:

- `default-param-last` was too annoying to fix
- `require-atomic-updates` was also too annoying to fix
- `import/no-unassigned-import` all the instances of this in the codebase are legitimate—I don't think we should enable it if every time it flags something we need to disable it

I'm open to enabling those in separate PRs in the future, should someone feel inclined.

### CODE REVIEW

File: app/scripts/controllers/threebox.js [hunk 1]
```diff
@@ -1,4 +1,9 @@
 import ObservableStore from 'obs-store'
+
+/* eslint-disable import/first,import/order */
```
Comment (Gudahtt) [inline app/scripts/controllers/threebox.js:3]: Are these rules disabled here because they conflicted with each other?

Comment (whymarrh) [reply app/scripts/controllers/threebox.js:3]: They're disabled here because they both apply to the following code

Comment (whymarrh) [reply app/scripts/controllers/threebox.js:3]: - `import/first` wants the `require` and `import`s to move
- `import/order` wants the `require` call alone to move

They're both disabled here and re-enabled as soon as ESLint will allow.

Comment (whymarrh) [reply app/scripts/controllers/threebox.js:3]: The e2e tests break if I move the require call so I don't want to move it

File: ui/app/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js [hunk 2]
```diff
@@ -125,13 +125,9 @@ export default class PermissionPageContainerContent extends PureComponent {
...
       )
     }
-    return t(
-      'connectTo',
-      [
-        this.getAccountDescriptor(selectedIdentities[0]),
-      ],
-    )
-
+    return t('connectTo', [
```
Comment (Gudahtt) [inline ui/app/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js:128]: What motivated this change 🤔

Comment (whymarrh) [reply ui/app/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js:128]: Whoops, habit—I'd meant to only remove the extra newline but got carried away

File: ui/app/helpers/utils/fetch-with-cache.js [hunk 3]
```diff
@@ -26,9 +26,9 @@ const fetchWithCache = async (url, fetchOptions = {}, { cacheRefreshTime = 36000
     return cachedResponse
   }
   fetchOptions.headers.set('Content-Type', 'application/json')
-  const _fetch = timeout ?
-    fetchWithTimeout({ timeout }) :
-    window.fetch
+  const _fetch = timeout
+    ? fetchWithTimeout({ timeout })
+    : window.fetch
```
Comment (whymarrh) [inline ui/app/helpers/utils/fetch-with-cache.js:31]: These changes are extraneous, not suggested by ESLint

File: app/scripts/migrations/002.js [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
-
 import { cloneDeep } from 'lodash'
 
+const version = 2
 
 export default {
   version,
 
-  migrate: function (originalVersionedData) {
+  migrate (originalVersionedData) {
```
Comment (Gudahtt) [inline app/scripts/migrations/002.js:8]: What syntax is this O_O

Comment (whymarrh) [reply app/scripts/migrations/002.js:8]: [Method definitions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions):

> Starting with ECMAScript 2015, a shorter syntax for method definitions on objects initializers is introduced. It is a shorthand for a function assigned to the method's name.

File: development/build/index.js [hunk 5]
```diff
@@ -41,8 +41,7 @@ function defineAllTasks () {
         manifestTasks.dev,
         reload,
       ),
-    ),
```
Comment (Gudahtt) [inline development/build/index.js:44]: This change is unexpected. I prefer the older format, and I'm not sure what could have caused this 🤔

File: test/lib/createTxMeta.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
 import { snapshotFromTxMeta } from '../../app/scripts/controllers/transactions/lib/tx-state-history-helpers'
 
 export default function createTxMeta (partialMeta) {
-  const txMeta = Object.assign({
+  const txMeta = {
     status: 'unapproved',
-    txParams: {},
-  }, partialMeta)
+    txParams: {}, ...partialMeta,
```
Comment (Gudahtt) [inline test/lib/createTxMeta.js:6]: It's weird to see multi-line object literals with things doubled up on the same line like this 🤔 I would have expected this to be disallowed.

File: development/build/index.js (hunk 5 above)
Comment (whymarrh) [reply development/build/index.js]: The first arg to the function is mis-aligned, causing the auto-fix to do this. I'll fix this with db7f5f200.

File: test/lib/createTxMeta.js (hunk 6 above)
Comment (whymarrh) [reply test/lib/createTxMeta.js]: I'm not sure why this is allowed. I've fixed it with 4ae9694cf.

File: ui/app/ducks/gas/gas-duck.test.js [hunk 7]
```diff
@@ -412,16 +404,17 @@ describe('Gas Duck', function () {
...
 
-      await fetchBasicGasAndTimeEstimates()(mockDistpatch, () => ({ gas: Object.assign(
-        {},
-        initState,
-        { basicPriceAndTimeEstimatesLastRetrieved: 1000000 },
-      ),
-      metamask: { provider: { type: 'ropsten' } },
+      await fetchBasicGasAndTimeEstimates()(mockDistpatch, () => ({
+        gas: {
+
```
Comment (Gudahtt) [inline ui/app/ducks/gas/gas-duck.test.js:409]: Nit: Extra newline here

```suggestion
```

File: test/lib/createTxMeta.js (hunk 6 above)
Comment (Gudahtt) [reply test/lib/createTxMeta.js]: The same problem is present in _many_ other places, but I'm not gonna block on it. Hopefully prettier will fix it.

File: development/build/index.js (hunk 5 above)
Comment (Gudahtt) [reply development/build/index.js]: The same problem is present in many other places, but I'm not gonna block on it. Hopefully prettier will fix it.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**